packages/react-ai/src/components/AIConversation/views/Controls/InputControl.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import debounce from 'lodash/debounce.js';
+
+import { withBaseElementProps } from '@aws-amplify/ui-react-core/elements';
+
+import { AIConversationElements } from '../../context/elements';
+import { InputContext } from '../../context';
+import { AttachFileControl } from './AttachFileControl';
+
+const { Button, Icon, TextArea, View } = AIConversationElements;
+
+const TYPEAHEAD_DELAY_MS = 300;
+
+const INPUT_BLOCK = 'ai-input';
+
+const sendIconProps = () => ({
+  children: (
+    <svg>
+      <g clipPath="url(#clip0_387_1426)">
+        <path
+          d="M14.3199 7.98214L1.66669 1.23486L3.97674 7.98214M14.3199 7.98214L1.66669 14.5689L3.97674 7.98214M14.3199 7.98214H3.97674"
+          strokeWidth="1.5"
+          strokeLinecap="square"
+          strokeLinejoin="round"
+        />
+      </g>
+      <defs>
+        <clipPath id="clip0_387_1426">
+          <rect width="100%" height="100%" />
+        </clipPath>
+      </defs>
+    </svg>
+  ),
+  stroke: 'white',
+  'aria-hidden': true,
+  className: `${INPUT_BLOCK}__icon`,
+  width: '24',
+  height: '24',
+  viewBox: '0 0 16 16',
+  fill: 'none',
+  xmlns: 'http://www.w3.org/2000/svg',
+});
+
+const SendIcon = withBaseElementProps(Icon, sendIconProps);
+
+const SendButtonBase = withBaseElementProps(Button, {
+  'aria-label': 'Send message',
+  className: `${INPUT_BLOCK}__button ${INPUT_BLOCK}__button--send`,
+  disabled: false,
+});
+
+const SendButton: typeof SendButtonBase = React.forwardRef(
+  function SendButton(props, ref) {
+    // TODO should come from context
+    const isWaitingForResponse = false;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars -- TODO send message on click
+    const { input } = React.useContext(InputContext);
+    return (
+      <SendButtonBase
+        onClick={() => {}}
+        disabled={isWaitingForResponse}
+        {...props}
+        ref={ref}
+      />
+    );
+  }
+);
+
+const TextInputBase = withBaseElementProps(TextArea, {
+  className: `${INPUT_BLOCK}__input`,
+  placeholder: 'Message Raven',
+  id: `${INPUT_BLOCK}-text-input`,
+});
+
+const TextInput: typeof TextInputBase = React.forwardRef(
+  function TextInput(props, ref) {
+    const { input, setInput } = React.useContext(InputContext);
+    // TODO should come from context or prop
+    const isFirstMessage = false;
+
+    const debouncedSetInput = debounce((value: string) => {
+      setInput!(value);
+    }, TYPEAHEAD_DELAY_MS);
```

Comment:
Debouncing is relevant for reducing network calls, but debouncing change vents makes the UI seem sluggish to end users. Should remove this, feel free to ping me offline if you want to discuss